DOCS-5916 adding contracts (#87)
mulesoft/docs-exchange: modules/ROOT/pages/to-request-access.adoc

endif::[]
:imagesdir: ../assets/images/

> When you request access, the client application is registered and creates an

Reviewer: To implement an API, the client application must first request access. When the request is approved by the API, a contract is created between the client application and the API, and the client application is registered.

Author: Updated to use this text. Thank you!

> Requesting access registers a client and creates an API contract between an API

Reviewer 2: When you request access, a client is registered and creates an API contract between an API

Author: Fixed.

> and a client application.

Reviewer 2: ```suggestion
and the client application.
```

> API contract between the API and the client.

Reviewer: creates a contract between the API and client application--not an API contract.

Author: Fixed.

> APIs that have been protected by a client ID enforcement policy require

Reviewer 2: ```suggestion
APIs that are protected by a client ID enforcement policy require
```

Reviewer 2: I assume the repetition of this text is intentional since it's two different files. Good opportunity for an include.  :-)

Author: Good point. It is intentional, and it is a good opportunity for an include...

> applications to have a client ID and client secret. This enables the

Reviewer: APIs that are protected by a Client ID Enforcement Policy require client application to provide the client ID and client secret credentials.

*the client ID and client secret credentials are automatically created when the client application is registered.*

Author: Added this info.

application to access the API and provides consumer analytics for an API
developer.

API contracts are visible to API consumers in Exchange, and visible to API
providers in API Manager.

API providers can view, manage, and revoke these contracts with the steps in
xref:api-manager::manage-client-apps-latest-task.adoc[Managing API Instance Contracts on API Manager].

> If terms and conditions are set for an asset, you are prompted to accept the terms before access can be granted.

Reviewer 2: ```suggestion
If terms and conditions are set for an asset, you are prompted to accept the terms before access is granted.
```

. Click an API asset to view more details.

> . Click Request Access.

Reviewer 2: ```suggestion
. Click **Request Access**.
```

> . If you do not have an application you want to provide access for, click Create A New Application in the Request Access menu. See the following section for more information.

Reviewer 2: ```suggestion
. If you do not have an application to which to provide access, click **Request Access > Create A New Application**. See the following section for more information.
```

endif::[]
:imagesdir: ../assets/images/

> To implement an API, the client application must first request access. When the

Reviewer 3: See previous comment.

Author: Fixed.

request is approved by the API, a contract is created between the client
application and the API, and the client application is registered.

APIs that are protected by a client ID enforcement policy require client

> applications to provide a client ID and client secret. The client ID and client

Reviewer 3: Secret is optional

Author: Fixed.

secret credentials are automatically created when the client application is

> registered. These credentials enable the application to access the API and

Reviewer 3: This is somewhat confusing since it seems to imply that only with this policy and this request analytics is enabled. Analytics is enabled for all tracked applications. In that case, they are generated even if no policy is applied.

Author: Fixed.

endif::[]
:imagesdir: ../assets/images/

To register a client application to an existing API, the client application
must first request access. When the request is approved by the API owner, a
contract is created between the client application and the API, and the client
application is registered.

APIs that are protected by a client ID enforcement policy require client
applications to provide a client ID and optional client secret. The client ID and client
secret credentials are automatically created when the client application is
registered.

An API owner can use API Manager to view all API contracts ever created for an
API instance. An API consumer can use Exchange's *My Applications* page to view
the API contracts created between an API instance and that consumer's client
applications.

> API providers can view, manage, and revoke these contracts with the steps in

Reviewer 4: ```suggestion
API providers can view, manage, and revoke these contracts by following the steps in
```

xref:api-manager::manage-client-apps-latest-task.adoc[Managing API Instance Contracts on API Manager].

> If terms and conditions are set for an asset, you are prompted to accept the terms before access is granted.

Reviewer 4: How are these related to a contract? Are we saying that if the contract includes terms and conditions...?

Reviewer 3: Terms & Conditions may be defined at organization level. If defined, then the contract stores the terms and conditions accepted when the contract was created.

. Click an API asset to view more details.

> . Click **Request Access**.

Reviewer 4: Here and below, why the double asterisks?   **   **

Reviewer 3: Please see: https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/#formatted-text

> . If you do not have an application to which to provide access, click **Request Access > Create A New Application**. See the following section for more information.

Reviewer 4: Suggest naming the section rather than saying "the following section," to keep the text independent of reshuffling.
